store/tikv/oracle/oracles/pd.go

```diff
@@ -294,6 +294,12 @@ func (o *pdOracle) getStaleTimestamp(txnScope string, prevSecond uint64) (uint64
 func (o *pdOracle) GetStaleTimestamp(ctx context.Context, txnScope string, prevSecond uint64) (ts uint64, err error) {
 	ts, err = o.getStaleTimestamp(txnScope, prevSecond)
 	if err != nil {
+		// If any error happened, we will try to fetch tso and set it as last ts.
+		ts, tsErr := o.getTimestamp(ctx, txnScope)
+		if tsErr != nil {
+			return 0, errors.Trace(tsErr)
+		}
+		o.setLastTS(ts, txnScope)
```

Comment:
I agree with @xhebox , this code block is just the same as `GetTimestamp`.

Reply:
updated

Reply:
I didn't see it's updated... I mean to replace this code block with a single `GetTimeStamp`.

Reply:
got it.